src/copy.c

```diff
@@ -85,6 +85,15 @@
  */
 #define MAX_BUFFERED_BYTES 65535
 
+/* Trim the list of buffers back down to this number after flushing */
+#define MAX_PARTITION_BUFFERS 32
+
```

Comment:
Why not keep all of them? This doesn't influence the max memory usage of the query, and saves the work of (de)allocating these buffers back and forth. I think max memory usage is the most important because this is what determines if the query fails with OOM. Transiently reducing the memory usage helps somewhat, but I'm not sure it's worth the work.

Reply:
We should definitely delete the buffers after some time. Imagine you have a copy operation from STDIN running which ingests data from a stream over days. The buffers for the older chunks are at some point no longer needed and should not lead to an OOM.

Reply:
Oh, they are actually fixed after a specific chunk... I had a wrong understanding of this. Then we have to remove them eventually indeed.